packages/jest-runtime/src/index.ts

```diff
@@ -208,6 +208,7 @@ export default class Runtime {
   private readonly esmConditions: Array<string>;
   private readonly cjsConditions: Array<string>;
   private isTornDown = false;
+  private isInsideTestCode: boolean | undefined;
```

Comment:
I am using value `undefined` as a sign that no test started yet so Jest can still use requires of modules prior running the test module (e.g. requiring the test module itself).